### PR TITLE
reset password functionality

### PR DESCRIPTION
**Pull request by nobane**
@justinnoah please review this one carefully as it has security implications.

### CODE REVIEW

**Comment by justinnoah** [comment]
Missing Flask-Mail in requirements.txt

**Comment by nobane** [comment]
Good catch, added that to requirements.txt.

**Comment by justinnoah** [comment]
Migrations fail with:

```
sqlalchemy.exc.CompileError: Postgresql ENUM type requires a name.
```

when executing:

```
PYTHONPATH="." alembic upgrade head
```

**Comment by nobane** [comment]
@justinnoah migrations were really screwy, please try again now.

File: _15thnight/models.py [hunk 1]
```diff
@@ -14,7 +18,7 @@ class User(Model):
     """
     User Model.
 
-    Required parameters:
+    Required paramets:
```
Comment (justinnoah) [inline _15thnight/models.py]: what?

File: _15thnight/models.py [hunk 2]
```diff
@@ -41,6 +47,10 @@ def check_password(self, password):
         """Check a user's password (includes salt)."""
         return check_password_hash(self.password, password)
 
+    def generate_reset_token(self):
+        self.reset_token = os.urandom(32).encode('base64')[:-2].replace('/', '')
```
Comment (justinnoah) [inline _15thnight/models.py]: UUID should work instead of this magic

File: requirements.txt [hunk 3]
```diff
@@ -3,12 +3,14 @@ amqp==1.4.9
 anyjson==0.3.3
 bcrypt==2.0.0
 billiard==3.3.0.23
+blinker==1.4
```
Comment (justinnoah) [inline requirements.txt]: What's this for? I don't see it being used anywhere; it is not a a Flask-Mail dependency

**Comment by justinnoah** [comment]
Nitpick: move the sign-in button below the forgot password link. OR only show the link after an incorrect login attempt.

**Comment by justinnoah** [comment]
:+1:

**Pull request merged**